extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/statistics/DistinctKeyCollector.java

```diff
@@ -43,8 +43,8 @@
  */
 public class DistinctKeyCollector implements KeyCollector<DistinctKeyCollector>
 {
-  static final int INITIAL_MAX_KEYS = 2 << 15 /* 65,536 */;
-  static final int SMALLEST_MAX_KEYS = 16;
+  static final int INITIAL_MAX_BYTES = 5_120_000;
```

Comment:
Should this be 10 MB?

Reply:
Also, I think we should bump this up to 300 MB in my view. We generally set 3 gig per worker so 10% of that should be a good start I think .